parser for French regions

AilinvandenHonigkuchenpferdlande opened the pull request:
- Doesn't include fetch_price (wrote to RTE - apparently only access to price data at fee?)
- Note on exchange: understood from instructions that zone_key1 -> zone_key2 ought to be ordered alphabetically, but I've ordered them by occurrence in the regional tables (see EXCHANGE dictionary).

AilinvandenHonigkuchenpferdlande commented:
I'd be happy to do those things!! Can someone tell me where to get the values for: 
- lonlat (exchange.json)
- rotation (exchange.json)
- bounding box (zones.json)

Also is it ok if the exchange zonekeys are ordered the way they are now (which made the most sense to me, but are not in alphabetical order)?

FelixDQ commented:
Thank you!

* lonlat (exchange.json): This is the location of the arrow. I would use a tool like this https://www.latlong.net/
* rotation (exchange.json): The rotation of the arrow (pointing from zone1 to zone2). 0 is north, 90 is east, 180 is south etc.  
* bounding box (zones.json): A bounding box around the region. I would use a tool like https://boundingbox.klokantech.com/ (pick CSV and copy the numbers)

Thanks for asking about the zone keys in the exchanges. I just checked with the exchanges and there are unfortunately a few things that depend on them being in alphabetical order.

Don't hesitate to ask if anything is unclear or if you have any other questions :-)

FelixDQ commented:
lonlat and rotation are also in this old file actually, so you can copy them from there:
https://github.com/tmrowco/electricitymap-contrib/blob/feature/hackathon/web/app/exchangeconfig.js

corradio commented:
@felixdq for the bounding box I think we have a script in the repo that does it. Unfortunately I don't have my laptop with me so I can't search for it :/

AilinvandenHonigkuchenpferdlande commented:
- [ ] zones.json: I've added the regions, including bounding boxes with the link @FelixDQ provided

- [ ] exchanges.json: I've fixed the alphabetical order of the exchanges parser and ready to add them to exchanges json. However, I noted that you have used other region codes before that are inconsistent with the ones I have found on RTE/ISO (https://en.wikipedia.org/wiki/ISO_3166-2:FR). I've researched the ones missing in my list (FR-ACA, FR-ALP, FR-CEN, FR-LRM, FR-NPP, FR-PLO) and can't find which region these abbreviations correspond to. From the 11 regions I've got, the following regional codes are missing from the exchanges list (FR-CVL, FR-GES, FR-HDF, FR-NAQ, FR-OCC, FR-PDL). What's the best way forward? Should I just pick lonlat coordinates and rotation for the arrow that I deem suitable? 

- [ ] geojson: any news here from @Kongkille ? @corradio shared this before but I don't know what to do with it! 
https://github.com/tmrowco/electricitymap-contrib/blob/feature/hackathon/web/app/FR_regions.json

Kongkille commented:
Hi @AilinvandenHonigkuchenpferdlande I'm currently working on splitting the file up for you so you have a single geojson for each zone! :)

Kongkille commented:
Alright, I've split the file into separate files, so you can upload.

The original file had some additional zones that are not covered in the parser, so I've removed those. I've named them corresponding to your FR_REGIONS object in the parser, so hopefully that should work.

I've also generated bounding boxes for each region, I can see you've done it also, but if you want to double-check it's here as well.

As for next steps, you should read this
https://github.com/tmrowco/electricitymap-contrib/blob/master/web/README.md

So basically, I think after importing the files in generate_geometries, you should then add them in the zone definitions. I don't think you need to do more than that. CC @corradio?

[FR.zip](https://github.com/tmrowco/electricitymap-contrib/files/5050607/FR.zip)

corradio commented:
If I remember correctly you also have to run the generate geometries script
once you've imported the geojson

On Mon 10 Aug 2020 at 13:48, Markus Killendahl <notifications@github.com>
wrote:

> Alright, I've split the file into separate files, so you can upload.
>
> The original file had some additional zones that are not covered in the
> parser, so I've removed those. I've named them corresponding to your
> FR_REGIONS object in the parser, so hopefully that should work.
>
> I've also generated bounding boxes for each region, I can see you've done
> it also, but if you want to double-check it's here as well.
>
> As for next steps, you should read this
> https://github.com/tmrowco/electricitymap-contrib/blob/master/web/README.md
>
> So basically, I think after importing the files in generate_geometries,
> you should then add them in the zone definitions. I don't think you need to
> do more than that. CC @corradio <https://github.com/corradio>?
>
> FR.zip
> <https://github.com/tmrowco/electricitymap-contrib/files/5050607/FR.zip>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/tmrowco/electricitymap-contrib/pull/2613#issuecomment-671308879>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMUIKCDGETIPEALVFJSADDR77NBZANCNFSM4PUPJ6XQ>
> .
>
-- 

Olivier Corradi

Founder, CEO

<https://www.linkedin.com/in/oliviercorradi> <https://twitter.com/corradio>

tmrow.com

FelixDQ reviewed (changes requested):
Nice job @AilinvandenHonigkuchenpferdlande !

I want to make a final review, but it looks like you accidentally committed a lot of file that shouldn't be committed, so it is a bit difficult for me to navigate. Can you revert the following?
   - These shouldn't be committed at all
       - [ ] build/*
       - [ ] node_modules/*
   - Changes to these files should be reverted
       - [ ] mockserver/public/v3/stat

FelixDQ commented on parsers/FR_R.py:
```diff
@@ -0,0 +1,310 @@
+
+def fetch(zone_key, session=None, target_datetime=None):
+    if target_datetime:
+        to = arrow.get(target_datetime, 'Europe/Paris')
```
```suggestion
        to = arrow.get(target_datetime, tz='Europe/Paris')
```

FelixDQ reviewed (changes requested):
Almost there! 🎉 

I added a few comments.

**Historical data**
When I'm running the test_parser.py with --target_datetime it returns an empty list. Do you know why? This parameter is important because we need to fill the database with historical data.

**Files**
Looks like there are few unintentional changes to files left. I think a small error happened when updating the branch. There should only be changes to the following files. 
* web/third_party_maps/*
* web/src/world.json
* web/generate-geometries.js
* parsers/FR_R.py
* config/exchanges.json
* config/zones.json

I think that this can be resolved by trying to merge master again. Let me know if I should try to fix it or if you need help.

**Next**
Also just a fyi. After this is done, then I need to do some validation to make sure that this doesn't break anything. We have some algorithms that are very sensitive to data quality. Also, there might be some issues that arise when a zone is split up. I haven't done this before, so it might take a few weeks unfortunately.

AilinvandenHonigkuchenpferdlande commented:
Hi @FelixDQ , thanks for the comments! I'll try to do it today. 

Regarding historical data: there seems to be no data before 1st July 2020. Hence why you can't fetch it, if you set it to --target_datetime 2018-01-01T08:00. However, try any date from the 1st July 2020, it should work (at least it did for me just now).

FelixDQ commented:
I fixed the merge conflicts from earlier :-)

> Regarding historical data: there seems to be no data before 1st July 2020. Hence why you can't fetch it, if you set it to --target_datetime 2018-01-01T08:00. However, try any date from the 1st July 2020, it should work (at least it did for me just now).

Ah bummer, this actually a bit unfortunate. We need at least a year of data to train a forecast which is used by the API. A lot of people are using the FR forecast, so this is actually blocking atm 😞 . Though I really want to merge your hard work, so I will investigate opportunities to proceed anyway. I currently see the following possibilities:
1. We find historical data from another source and fill our database with that. (We would need to compare the data with the data from the API)
2. We wait a year before merging this 😨 
3. I find some way to include the regions in the live map and keep using FR for the API. I need to investigate the complexity of this, but this would be really nice as I don't like blocking PR's because of dependencies by the API.'

If anyone reading this knows a good source for `1`, then please let us know. Otherwise I will look into `3`

Kongkille commented:
@FelixDQ I think we should discuss how 3. can work in the future, perhaps we can build a simple infrastructure that allows this, so if we **know** that a data source will be better than the current one (in this case for example), but just doesn't have the historic timespan, it will be easy to switch it over to be the API version after a while?

AilinvandenHonigkuchenpferdlande commented:
@FelixDQ thank you! FYI, I have just sent an email to RTE to inquire 1) whether they have any historical data (they might not have uploaded it because not clean or whatever reason) beyond 01.07.2020 and 2) if they know any other sources we could access. Will keep you up to date on any progress. Would be a shame to wait a year :O

AilinvandenHonigkuchenpferdlande commented:
@FelixDQ , RTE responded to me and said they have consolidated regional data here (it's hourly) from 2013 to 2020, which is also accessible via the API... 

https://opendata.reseaux-energies.fr/explore/dataset/eco2mix-regional-cons-def/information/?disjunctive.libelle_region&disjunctive.nature

If you search some of the more recent data starting from 2018, the breakdown is the same as the regional data right now. 

What do you suggest we do with this historical data?

FelixDQ commented:
Awesome! Does this include exchange data as well?

Ideally, we want the parser to be able to fetch all the data needed. This could be achieved by adding a conditional that fetches from this other endpoint if the date is earlier than `x`, where `x` is the earliest date we have data from the original source.

Also, one thing I noticed is that the `x` mentioned above seems to change. I can for example no longer fetch data for 1st July 2020.

Does this make sense?

AilinvandenHonigkuchenpferdlande commented:
It should exclude exchange data (the columns are there, but there seem to be no numbers... only 'ND') - will have to check on that with RTE, and also ask them whether any major changes happen when they consolidate the data

Question to you: would electricitymap also have to update the real-time data saved if the consolidated numbers are slightly different? 

Also just found out that they consolidate the real-time data once a month, which is when it gets moved from the 'real-time dataset' to the 'consolidated dataset'. So yes, that should explain why you can't access data from 1st July anymore.

Let me figure out the exchange data situation and then add the conditional for historical data. Will get in touch if I have any questions!

FelixDQ commented:
> Question to you: would electricitymap also have to update the real-time data saved if the consolidated numbers are slightly different?

No, we currently only fetch historical data manually.

Hopefully, there isn't any significant difference between the two datasets, as inaccuracies will be multiplied in the forecast and marginal.

FlorentVTT commented:
Hello All. Interested also to see the FR region in electricitymap :) I dont know what blocks now.
But i have also a simple Question : As Regional datas does not make difference between Thermal source (among fioul, gaz, coal), how it would be possible to rely only on Regional data for FR ?

AilinvandenHonigkuchenpferdlande commented:
To integrate it, we have to have 1 year worth of historical data. I have been looking into the consolidated dataset (which goes back to 2013). However, the exchange data that is needed only became available starting from January 2020.

corradio commented:
Turns out most interconnectors are missing at this stage. It seems they are not available in the API, which is weird.

corradio commented:
I've started working a bit further on this internally.
However, two things are preventing us to move further here:
1. we're missing most of the interconnector data (we need to talk with the data provider)
2. our infrastructure doesn't currently have a simple way to migrate from country data to regional data. This is work we plan to do

Kongkille commented:
@corradio have you reached out to the data provider? Otherwise, we might want to create a subissue or track it somewhere related to the PR

corradio commented:
> @corradio have you reached out to the data provider? Otherwise, we might want to create a subissue or track it somewhere related to the PR

I tweeted them but no response. 
I think we should set make sure someone reaches out and track it in the issue.

FYI I had done some more work on top of this PR on this branch: https://github.com/tmrowco/electricitymap-contrib/tree/olc/split-FR
I fixed some things regarding zone configurations, parser datetimes and geometries.
I'm wondering if we could close this PR and keep a reference to the branch while we wait for the data provider?

AilinvandenHonigkuchenpferdlande commented:
@corradio I did reach out to the data provider a couple times on data questions. They usually respond within a couple days. The central contact point is 

rte-opendata@rte-france.com

The team member responding to my requests on the France regional data was Nathalie Larmet: nathalie.larmet@rte-france.com

super happy to see you’re picking it up again :)

corradio commented:
Great I will reach out to them!

On Wed, Mar 3, 2021 at 4:15 PM Ailin Huang <notifications@github.com> wrote:

> @corradio <https://github.com/corradio> I did reach out to the data
> provider a couple times on data questions. They usually respond within a
> couple days. The central contact point is
>
> rte-opendata@rte-france.com
>
> The team member responding to my requests on the France regional data was
> Nathalie Larmet: nathalie.larmet@rte-france.com
>
> super happy to see you’re picking it up again :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/tmrowco/electricitymap-contrib/pull/2613#issuecomment-789788043>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMUIKAXZFWGZ3Y6TTA6XZ3TBZHCPANCNFSM4PUPJ6XQ>
> .
>


-- 

Olivier Corradi

Founder, CEO

<https://www.linkedin.com/in/oliviercorradi> <https://twitter.com/corradio>

tmrow.com

corradio commented:
Hi all,
I think it makes more sense to close this PR and to use issue https://github.com/tmrowco/electricitymap-contrib/issues/2777 to track progress. The work here is not lost as it was saved to a branch, from which we can work from as soon as have obtained the data.
Olivier
